### PR TITLE
CI

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-gh-pages-explorer.yml:31]: I think it's nice if we always describe what a step does, makes the logs look prettier ✨

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-gh-pages-explorer.yml:9]: Isn't it safe to always use the [latest aliases](https://docs.alephium.org/dapps/public-services#api-aliases)?

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-gh-pages-explorer.yml:41]: Notice the `ci:build:gh-pages`. It will be explained later in the PR.

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-gh-pages-explorer.yml:1]: I have not yet tested this workflow.

File: .github/workflows/publish-docker-explorer.yml
Comment (nop33) [inline .github/workflows/publish-docker-explorer.yml:5]: This is the format that `bunx changeset tag` creates: `[package-name]@X.X.X`. I think it makes sense.

File: .github/workflows/publish-docker-explorer.yml
Comment (nop33) [inline .github/workflows/publish-docker-explorer.yml:37]: Now that the tags are prefixed with the package name (still discussing this on Slack), instead of figuring out how to extract the version from the tag, I decided to use an action that gets the version from the package.json file.

File: .github/workflows/publish-docker-explorer.yml
Comment (nop33) [inline .github/workflows/publish-docker-explorer.yml:52]: That's the var that `martinbeentjes/npm-get-version-action` exploses.

File: .github/workflows/deploy-mainnet-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-mainnet-explorer.yml:1]: I have not yet tested this workflow.

File: .github/workflows/deploy-testnet-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-testnet-explorer.yml:1]: I have not yet tested this workflow.

File: .github/workflows/publish-docker-explorer.yml
Comment (nop33) [inline .github/workflows/publish-docker-explorer.yml:1]: I have not yet tested this workflow.

File: .github/workflows/release-desktop.yml
Comment (nop33) [inline .github/workflows/release-desktop.yml:1]: I have not yet tested this workflow.

File: .github/workflows/release-desktop.yml
Comment (nop33) [inline .github/workflows/release-desktop.yml:50]: Since `samuelmeuli/action-electron-builder` will execute the `build` script by default, we don't have a chance to run the `compile` script of the `shared` package. See later how `ci:build` fixes this.

File: .github/workflows/static-analysis.yml
Comment (nop33) [inline .github/workflows/static-analysis.yml:1]: This workflow I DID test and it works ✨ Waiting for bun to release a new version to include auditing.

File: .github/workflows/test.yml
Comment (nop33) [inline .github/workflows/test.yml:1]: This workflow also was tested and works! We just need to figure out windows...

File: apps/desktop-wallet/README.md
Comment (nop33) [inline apps/desktop-wallet/README.md:22]: ☝️ This.

File: apps/desktop-wallet/package.json
Comment (nop33) [inline apps/desktop-wallet/package.json:12]: `homepage` and `repository` are needed for `electron-builder` to work.

`private` was added since the package is not published on npm.

File: apps/desktop-wallet/package.json
Comment (nop33) [inline apps/desktop-wallet/package.json:24]: To ensure that the `compile` task of every dependency of this app will run before the `build` does, we execute the turbo `build*` commands. See the workspace-specific `turbo.json` below.

FYI, we can have many turbo configs, like so:

```
./turbo.json
./apps/desktop-wallet/turbo.json
./apps/explorer/turbo.json
```

Very useful feature.

See more: https://turbo.build/repo/docs/core-concepts/monorepos/configuring-workspaces

File: apps/desktop-wallet/turbo.json
Comment (nop33) [inline apps/desktop-wallet/turbo.json:2]: This means that it extends the root `turbo.json`

File: apps/desktop-wallet/package.json
Comment (nop33) [inline apps/desktop-wallet/package.json:20]: We can remove `bun run build`, and manage this through Turborepo. See below.

File: apps/desktop-wallet/turbo.json
Comment (nop33) [inline apps/desktop-wallet/turbo.json:23]: Here we say that each `build:electron*` task requires the `build` task to run first. The `build` task is defined in the root `turbo.json`.

File: apps/desktop-wallet/vite.config.ts
Comment (nop33) [inline apps/desktop-wallet/vite.config.ts:59]: Also mentioned in https://vitejs.dev/guide/dep-pre-bundling.html#monorepos-and-linked-dependencies

File: apps/explorer/package.json
Comment (nop33) [inline apps/explorer/package.json:5]: This is also not published on npm, thus private

File: apps/explorer/package.json
Comment (nop33) [inline apps/explorer/package.json:11]: Once again, to ensure that `compile` is run, we use Turborepo

File: apps/explorer/vite.config.ts
Comment (nop33) [inline apps/explorer/vite.config.ts:1]: Quite similar config as in the desktop wallet

File: package.json
Comment (nop33) [inline package.json:14]: These commands are mainly needed so that the CI can run them. But, of course, we can also run them locally, even though we will have turbo available locally anyway.

File: packages/shared/package.json
Comment (nop33) [inline packages/shared/package.json:10]: It was a pain in the 🍑 to figure out polyfills, so I just reverted to exactly the same config we had before with the SDK, when everything was working fine. As discussed, we need to review the code in the shared package and strip away things that might not be necessary, review our dependencies, etc.

File: turbo.json
Comment (nop33) [inline turbo.json:12]: This is where the magic happens. Here we say that _"Before executing the `build` script of a package, make sure that the `compile` task OF ITS DEPENDENCIES (that's what the `^` means) is executed first"._

So, for the desktop wallet, for example, the `shared` package is listed in its dependencies array. Thus, when running `turbo run build` (or `bun run ci:build`), the `compile` script of the shared package will run first. ✨

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (nop33) [reply .github/workflows/deploy-gh-pages-explorer.yml:1]: I am not sure how to trigger this workflow to test it. Since it's not merged yet, it doesn't show up in GH's UI to press the button.

I'll see if I can do it locally with `act`.

Comment (nop33) [reply .github/workflows/deploy-gh-pages-explorer.yml:1]: Everything worked with `act` up until the "Set up Pages" step:

```
::error::Get Pages site failed. Please verify that the repository has Pages enabled and configured to build using GitHub Actions, or consider exploring the `enablement` parameter for this action.
::error::Error: Parameter token or opts.auth is required
```

It appears we need to change something in the settings of the GH repo?

Comment (nop33) [reply .github/workflows/deploy-gh-pages-explorer.yml:1]: I compared with the previous explorer repo and I think once we merge the workflows there will be some more info added in the settings page of the monorepo:

image

File: .github/workflows/deploy-testnet-explorer.yml
Comment (nop33) [inline .github/workflows/deploy-testnet-explorer.yml:1]: I've tested the workflow with `act` and it worked well until the step "Sync files to S3":

```
exitcode '127': command not found, please refer to https://github.com/nektos/act/issues/107 for more information
```

but hey, everything up until there worked :D ✨

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (mvaivre) [reply .github/workflows/deploy-gh-pages-explorer.yml:1]: Let's see!

File: .github/workflows/deploy-gh-pages-explorer.yml
Comment (mvaivre) [reply .github/workflows/deploy-gh-pages-explorer.yml:9]: Mhhh, I think it's ok if the testnet deploy fails, but I would use fixed version for the mainnet version.

File: .github/workflows/publish-docker-explorer.yml
Comment (mvaivre) [reply .github/workflows/publish-docker-explorer.yml:37]: Makes sense.

File: turbo.json
Comment (mvaivre) [reply turbo.json:12]: All clear 🫡

**Review by mvaivre — APPROVED** [review]
Super clear to me! Looks pretty 🤙 ✨

**Pull request merged**
